pkg/logs/logs.go

```diff
@@ -1,8 +1,164 @@
+	// logging flags, even those that are hidden.
+	configuration = logsapi.NewLoggingConfiguration()
+	// Logging features will be added to this feature gate, but the
+	// feature-gates flag will be hidden from the user.
+	features = featuregate.NewFeatureGate()
 )
 
-var Log = log.New(os.Stderr, "", log.LstdFlags)
+func init() {
+	runtime.Must(logsapi.AddFeatureGates(features))
+	// Turn on ALPHA options to enable the split-stream logging options.
+	runtime.Must(features.OverrideDefault(logsapi.LoggingAlphaOptions, true))
+}
+
+// AddFlags adds log related flags to the supplied flag set.
+//
+// The default logging format is changed to JSON. The default in Kubernetes
+// component base is "text", for backwards compatibility, but that is not a
+// concern for venafi-kubernetes-agent.
+// The split-stream options are enabled by default, so that errors are logged to
+// stderr and info to stdout, allowing cloud logging systems to assign an
+// severity INFO or ERROR to the messages.
+func AddFlags(fs *pflag.FlagSet) {
+	var tfs pflag.FlagSet
+	logsapi.AddFlags(configuration, &tfs)
+	features.AddFlag(&tfs)
+	tfs.VisitAll(func(f *pflag.Flag) {
+		if !visibleFlagNames.Has(f.Name) {
+			tfs.MarkHidden(f.Name)
+		}
+
+		// The original usage string includes details about how
+		// JSON logging is only available when BETA logging features are
+		// enabled, but that's not relevant here because the feature is enabled
+		// by default.
+		if f.Name == "logging-format" {
+			f.Usage = `Sets the log format. Permitted formats: "json", "text".`
+		}
+		if f.Name == "log-text-split-stream" {
+			f.DefValue = "true"
+			runtime.Must(f.Value.Set("true"))
+		}
+		if f.Name == "log-json-split-stream" {
+			f.DefValue = "true"
+			runtime.Must(f.Value.Set("true"))
+		}
+
+		// Since `--v` (which is the long form of `-v`) isn't the standard in
+		// our projects (it only exists in cert-manager, webhook, and such),
+		// let's rename it to the more common `--log-level`, which appears in
+		// openshift-routes, csi-driver, trust-manager, and approver-policy.
+		// More details at:
+		// https://github.com/jetstack/jetstack-secure/pull/596#issuecomment-2421708181
+		if f.Name == "v" {
+			f.Name = "log-level"
+			f.Shorthand = "v"
+		}
+	})
+	fs.AddFlagSet(&tfs)
+}
+
+// Initialize uses k8s.io/component-base/logs, to configure the following global
+// loggers: log, slog, and klog. All are configured to write in the same format.
+func Initialize() {
+	// This configures the global logger in klog *and* slog, if compiled with Go
+	// >= 1.21.
+	logs.InitLogs()
+	if err := logsapi.ValidateAndApply(configuration, features); err != nil {
+		fmt.Fprintf(os.Stderr, "Error in logging configuration: %v\n", err)
+		os.Exit(2)
+	}
+
+	// Thanks to logs.InitLogs, slog.Default now uses klog as its backend. Thus,
+	// the client-go library, which relies on klog.Info, has the same logger as
+	// the agent, which still uses log.Printf.
+	slog := slog.Default()
+
+	Log = &log.Logger{}
```

Comment:
Can we just use `slog.SetDefault(...)` here?
Using `slog.New(logr.ToSlogHandler(logger))` we can convert the klog logger to an slog logger.

Reply:
Maybe. I'll consider that in a follow up.